### PR TITLE
fix(Section): fix spacing + rewrite to TypeScript

### PR DESCRIPTION
**Pull request by tujoworker**

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/components/section/properties.md [hunk 1]
```diff
@@ -4,12 +4,12 @@ showTabs: true
...
+| Properties                                  | Description                                                                                                                                                                                                                                    |
+| ------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `style_type`                                | _(optional)_ to define the style of the visual helper. Use and `Style ID` from below. Defaults to `mint-green-12`.                                                                                                                             |
+| `spacing`                                   | _(optional)_ will add the default spacing around the wrapped content. Use `large`, `medium` or `small`. Defaults to `false`. If `true`, then `spacing-default` is used. Se the [available sizes](/uilib/usage/layout/spacing#spacing-helpers). |
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/section/properties.md:10]: I tried following the link as described here, https://eufemia.dnb.no/uilib/usage/layout/spacing/#spacing-helpers, to try to understand what `spacing-default` is. I wasn't able to find anything about `spacing-default` or `default` there. Probably just nitpicking by me though 🤔

File: packages/dnb-eufemia/src/components/section/Section.tsx [hunk 2]
```diff
@@ -14,32 +13,54 @@ import {
...
+export type SectionSpacing = boolean | 'small' | 'medium' | 'large'
+
+export type SectionProps = {
+  /**
+   * To define the style of the visual helper. Use and `Style ID` from below. Defaults to `mint-green-12`.
+   */
+  style_type?: SectionStyleTypes
+
+  /**
+   * Will add the default spacing around the wrapped content. Use `large`, `medium` or `small`. Defaults to `false`. If `true`, then `spacing-default` is used. Se the <a href="/uilib/usage/layout/spacing#spacing-helpers">available sizes</a>.
```
Comment (langz) [inline packages/dnb-eufemia/src/components/section/Section.tsx:42]: Same topic https://github.com/dnbexperience/eufemia/pull/1573/files#r976418986

**Review by langz — APPROVED** [review]
Nice nice, I like the introduction of `SectionStyleTypes` and using it where it's needed. 
Makes it a lot easier to understand that the style_type is the same "thing"/prop in different components now, compared to when the style_type was just of type string 🤩 

I see that there is quite a few changes to the screenshot snapshots 📸 , consider if this is as expected or not 😺

File: packages/dnb-design-system-portal/src/docs/uilib/components/section/properties.md (hunk 1 above)
Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/components/section/properties.md:10]: No, you are right – this should be removed 🙏

**Comment by tujoworker** [comment]
> Nice nice, I like the introduction of `SectionStyleTypes` and using it where it's needed. Makes it a lot easier to understand that the style_type is the same "thing"/prop in different components now, compared to when the style_type was just of type string 🤩
> 
> I see that there is quite a few changes to the screenshot snapshots 📸 , consider if this is as expected or not 😺

yes, they are good. I changed the spacing in the examples.

**Pull request merged**